### ISSUE TITLE
Loops capture text outside of tags

### ISSUE THREAD

**Issue by jmarks-joshua** [alonrbar/easy-template-x#92]
With a word document containing 

> The intervener(s) are {#interveners}{name}, {/}

 and data of 
`
{
 interveners: [
      {name: 'Test 1'},
      {name: 'Test 2'}
]
}
`

my output is
> The intervener(s) are Test 1,
> The intervener(s) are Test 2,

Is there any way to do this and keep it all on one line and without copying the text before my opening loop tag?

**Comment by thegnuu** [comment]
The easiest way to achieve this is to properly prepare your string when generating your template data, try something like this:

```js
{
  intervenerString: [ {name: 'Test 1'}, {name: 'Test 2'} ].map((o) => o.name).join(', ') 
}
```

```
The intervener(s) are {intervenerString}
```

This way you will get rid of the trailing comma as well :)
Maybe you can event generate the complete string to get rid of the `(s)` as well, since you can count them....

**Comment by thegnuu** [comment]
Oh yeah, in this case, this won't work. We stopped allowing our customers to modify the documents themselves since it caused more issues than it solved, unfortunately ;)

But it should not be that hard to add your own plugin to achieve this.

We had the idea to use `angularFilters` from https://github.com/alonrbar/easy-template-x-angular-expressions and append helper functions, like `{ interveners | combine("name", ", ") }`, to add some functions which can be called for non-tech users since this seemed "easier" than to add multiple loop variants that behave differently with different prefixes or something, but we dropped the idea since we currently make the changes ourselves. But maybe this would be a way for you to achieve your goals. It will need some work on the resolver but should be possible and quite powerful.

**Comment by jmarks-joshua** [comment]
It's certainly an improvement, but in an ideal world I'd actually be able to construct more of the text in the loop. This solution doesn't allow `{#list}{name}: {dob}{#needsSeparator}, {/needsSeparator}{/list}` which is really what I was after. So I may still have to go down a more complex route.

Anyway thanks for the help and suggestions.

**Comment by alonrbar** [comment]
Was this issue happening inside a table by any chance?

**Comment by alonrbar** [comment]
Fixed in v4.0.0
